src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2298,22 +2320,29 @@ public String getString()
         this.expr.addLazyFunction("diamond", -1, (c, t, lv)->
         {
             CarpetContext cc = (CarpetContext)c;
-            if (lv.size() != 3 && lv.size() != 4 && lv.size() != 5)
+            if (lv.size() <1 || lv.size() > 5)//Can be anywhere between 1 and 5 args
             {
-                throw new InternalExpressionException("'diamond' region should be specified with 3 to 5 coordinates");
+                throw new InternalExpressionException("'diamond' region should be specified with 3 to 5 arguments");
             }
 
+            int blockposarg=2;//Checks if the first arg is a block pos triple
+
+            if(lv.get(0).evalValue(c) instanceof NumericValue)blockposarg=0;
+
+            BlockPos cpos=BlockArgument.findIn((CarpetContext)c,lv,0).block.getPos();
```

Comment:
findIn should raise appropriate exceptions then use offset to determine the remaining arguments, and raise exceptions as needed